fix(ci): Release workflowにcontents: write権限を追加

## 概要

Release Workflowでバイナリアセットのアップロードが失敗する。

## エラー内容

```
⚠️ Unexpected error fetching GitHub release for tag refs/tags/v0.1.1: HttpError: Resource not accessible by integration
##[error]Resource not accessible by integration
```

## 原因

`softprops/action-gh-release@v1` がリリースにアセットをアップロードする際、`GITHUB_TOKEN` の権限が不足。

GitHub Actionsではデフォルトで `GITHUB_TOKEN` には `contents: read` 権限しかなく、リリースへの書き込みには `contents: write` 権限が必要。

## 修正内容

`.github/workflows/release.yml` に `permissions` を追加:

```yaml
permissions:
  contents: write
```

## 再現手順

1. タグをプッシュ: `git tag v0.1.1 && git push origin v0.1.1`
2. Release Workflowが起動
3. "Upload binaries to release" ステップで失敗

## 影響

- v0.1.1 リリースにバイナリアセットがアップロードされていない

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -5,6 +5,9 @@ on:
     tags:
       - 'v*.*.*'
 
+permissions:
+  contents: write
+
 jobs:
   build:
     strategy:

